### ISSUE TITLE
[Bug] XMPlay does support subsongs

### ISSUE THREAD

**Issue by Erquint** [vgmstream/vgmstream#974]
The "Usage" section of documentation states following:
> XMPlay cannot support subsongs due to player limitations (with any plugin)

I must protest.
Using XMP-ZXTUNE (and an AIMP skin, don't get fooled by it), I recorded this GIF to demonstrate that the player seems to support subsongs with at least one other plugin:
![screengrab](https://i.imgur.com/NtFSKo8.gif)

**Comment by bnnm** [comment]
I'll tweak the wording, as by "subsongs" it means "vgmstream's kind of subsongs" and "(with any plugin)" means "(with any of vgmstream's plugins, *xmp-vgmstream* or *in_vgmstream*)".

[See here](https://github.com/vgmstream/vgmstream/issues/619), but (AFAIK) XMP's subsongs are basically .cue "subsections" that must share sample rate+channels and only differ in "time", and aren't true subsongs in vgmstream's and other players's sense (free to mix anything, like 44100 stereo then 32000 mono). XMPZXTUNE would have this limitation too.

If there is a way to support those subsongs I couldn't find any open source plugins or API to do so.

**Comment by dr0biwan** [comment]
> This could possibly be done better, possibly by implementing a dedicated subsong field like foobar does.

Is there an indication of subsong status from the library for a given file ? I probably should have looked before asking but if not then that'd be useful to be able for my wacup needs to then distinguish between things so files with sub-songs can be treated as a virtual playlist (technically something that later versions of winamp could do by providing a custom playlist reader service without all of the 2.x era 'hacks').

-dro

**Comment by dr0biwan** [comment]
I know winamp is dead which is why I'm being "the wacup guy" which was why I was asking (not that any local dedicated player has any true relevance in the mobile/streaming-centric world that we know exist within).

I can't remember when it was added but in some of the later winamp sdks there was the plLoadEx folder which showed what was needed to make a custom playlist handler (it doesn't have to be done as a dedicated w5s & could be done as part of the initialisation within an in_*.dll if running in a true/compatible winamp.

Sadly this information wasn't made as clear at the time it was added since it'd have been useful for 3rd party plug-ins to avoid doing the hacks that were needed (think it was around 5.3 or 5.5 but I'm not 100%).

I've already got the means of using a custom playlist handler with wacup (seeing as I had to implement my own compatible solution) along with checking if the file is expected to have subsongs or not as I do for SID & archives along with actual playlists.

So what's the best way to query if there are subsongs for a file with libvgmstream.

-dro
